public/js/eval_control.jsx

```diff
@@ -0,0 +1,189 @@
+class VoteGroup extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {selectedCandidate: null};
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleClick(selectedCandidate) {
+    this.setState({selectedCandidate: selectedCandidate})
+    $("#submit_button").removeClass("disabled");
+  }
+
+  handleSubmit() {
+    var ref = defaultDatabase.ref('/tokenization_votes/' + userUID + "/" + questionNumber + "/");
+
+    ref.set(this.state.selectedCandidate);
+
+    location.reload();
+  }
+
+  render() {
+    var data = this.props.data
+
+    var keys = Object.keys(this.props.data).filter(function (key) {
+      return key.startsWith("tokenization") && data[key]
+    });
+
+    var thisPointer = this;
+
+    var answers = keys.map(function (key) {
+      return <Answer key={key}
+                     answerKey={key}
+                     answer={data[key]}
+                     clickAction={thisPointer.handleClick.bind(thisPointer, key)}
+                     current={thisPointer.state.selectedCandidate}
+      />
+    })
+
+    defaultDatabase.ref('/tokenization_votes/' + userUID).once('value').then(function (snapshot) {
+      var prevAnswers = snapshot.val();
+      var prevAnswerSize = Object.keys(prevAnswers).length;
+
+      if (prevAnswers[questionNumber] && prevAnswerSize < 3000) {
```

Comment:
This part causes a null pointer exception at the very first time.
I think the variable prevAnswers should have a JSON object as a default value.
```
line 40 |   var prevAnswers = snapshot.val() || {};
```